pallets/score-staking/src/lib.rs

```diff
@@ -477,41 +431,70 @@ pub mod pallet {
 
 		#[pallet::call_index(9)]
 		#[pallet::weight((195_000_000, DispatchClass::Normal))]
-		pub fn update_token_staking_amount(
+		pub fn distribute_rewards(
 			origin: OriginFor<T>,
-			account: T::AccountId,
-			amount: BalanceOf<T>,
 			round_index: RoundIndex,
+			id_graphs_staking: Vec<(T::AccountId, BalanceOf<T>)>,
 		) -> DispatchResultWithPostInfo {
 			let _ = T::TEECallOrigin::ensure_origin(origin)?;
 
-			match Scores::<T>::get(&account) {
-				Some(mut s) => {
-					if round_index <= s.last_token_distributed_round {
-						return Err(Error::<T>::TokenStakingAmountAlreadyUpdated.into());
-					}
-					let last_round = Round::<T>::get();
-					s.last_round_reward += amount;
-					s.total_reward += amount;
-					s.unpaid_reward += amount;
-					s.last_token_distributed_round = last_round.index;
-					Scores::<T>::insert(account.clone(), s);
-					Self::deposit_event(Event::TokenStakingAmountUpdated {
-						account: account.clone(),
-						amount,
-					});
-					Ok(Pays::No.into())
-				},
-				None => Err(Error::<T>::UserNotExist.into()),
+			ensure!(
+				round_index > LastTokenDistributionRound::<T>::get(),
+				Error::<T>::RoundRewardsAlreadyDistributed
+			);
+
+			let id_graphs_staking_map: BTreeMap<T::AccountId, BalanceOf<T>> =
+				id_graphs_staking.into_iter().collect();
+
+			ensure!(
+				id_graphs_staking_map.len() <= T::MaxIDGraphAccountsPerCall::get() as usize,
+				Error::<T>::MaxIDGraphAccountsPerCallReached
+			);
+
+			let round_reward: BalanceOf<T> = (T::YearlyInflation::get() * T::YearlyIssuance::get()
+				/ YEARS.into()) * Self::round_config().interval.into();
+			let round_reward_u128 = round_reward.saturated_into::<u128>();
+
+			let total_stake_u128 = ParaStaking::Pallet::<T>::total().saturated_into::<u128>();
+			let total_score = Self::total_score();
+			let n = Self::round_config().stake_coef_n;
+			let m = Self::round_config().stake_coef_m;
+
+			for (a, mut p) in Scores::<T>::iter() {
+				let default_staking = BalanceOf::<T>::zero();
+				let id_graph_staking = id_graphs_staking_map.get(&a).unwrap_or(&default_staking);
```

Comment:
Shall we skip (continue) if `&a` is not found?

Reply:
All this code will not be needed in the new implementation 👍🏼